### PR TITLE
Libs 27 move authz to checker

### PR DESCRIPTION
**Pull request by JeremyGibson**
Re-imagines AuthChecker a central library that can provide library access to Authentication and Authorization in a modular way.

### CODE REVIEW

File: auth_checker/plugins/authorize/casbin_pl.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        super().__init__(*args, **kwargs)
+        adapter = kwargs.get("adapter", ADAPTER_MAP[CASBIN_AUTHORIZER_POLICY_ADAPTER]())
+        self.enforcer = casbin.Enforcer(model=model, adapter=adapter, enable_log=False)
+
+    def authorize(self, *args, **kwargs) -> bool:
+        """
+        Authorize a user to perform an action on an object
+        kwargs are not supported for casbin authorize
+        :param args:
+        :param kwargs:
```
Comment (lsanchez44) [inline auth_checker/plugins/authorize/casbin_pl.py:71]: should this be here if kwargs arent supported?

Comment (JeremyGibson) [reply auth_checker/plugins/authorize/casbin_pl.py:71]: Good question. I think they need to be there for the inherited class definition, but let me check again.

File: envrc_sample [hunk 2]
```diff
@@ -1,5 +1,17 @@
 layout pyenv 3.10.4
 
-dotenv_if_exists
+export AUTHORIZER=casbin_authorizer
+# If you need a specific model file, you can set it here
```
Comment (lsanchez44) [inline envrc_sample:4]: should we even mention the file documentation if the ability to use a rbac file doesn't exist yet? or maybe we could put a disclaimer here saying that functionality is coming

**Review by lsanchez44 — APPROVED** [review]
massive undertaking! had a few small questions comments but overall looking good

**Pull request merged**